addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/FullPathDisclosureScanRule.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.htmlparser.jericho.Source;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class FullPathDisclosureScanRule extends PluginPassiveScanner {
+    private static final int pluginID = 110009;
+    private static final String MESSAGE_PREFIX = "pscanalpha.fullpathdisclosurealert.";
+    private static final Logger LOGGER = LogManager.getLogger(FullPathDisclosureScanRule.class);
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A05_SEC_MISCONFIG,
+                    CommonAlertTag.WSTG_V42_ERRH_01_ERR);
+
+    @Override
+    public int getPluginId() {
+        return pluginID;
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        final String responseBody = msg.getResponseBody().toString();
+        String evidence;
+        // matches the following paths:
+        // example for windows : C:\folder\folder
+        // example for unix : /dir/dir/
+        // default paths : /usr or /home etc..
+        Pattern pathPattern =
+                Pattern.compile(
+                        "(?i)(/([a-z0-9]+/)+)|(([a-z0-9]:\\\\)+([a-z0-9]+\\\\)+)|/bin|/usr|/mnt|/proc|/sbin|/dev|/lib|/tmp|/opt|/home|/var|/root|/etc|\\\\Applications|\\\\Volumes|\\\\System|\\\\Users|\\\\Developer|\\\\Library");
+        Matcher pathMatcher = pathPattern.matcher(responseBody);
+
+        if (getHelper().isSuccess(msg)) {
+            return;
+        }
```

Comment:
This may as well be the first thing done in the method. No point instantiating local cars etc or they won't be used.

Reply:
done